### ISSUE TITLE
Document Information

### ISSUE THREAD

**Issue by dbell154** [AndBible/and-bible#1804]
**Is your feature request related to a problem? Please describe.**
When looking through download documents there is no clear description of each document. Long pressing and selecting the information icon does show the information I'd like to see.

**Describe the solution you'd like**
I think it would be better if the information (and icon) is accessible from the main list without long pressing. It would fit easily below the source name.

**Comment by tuomas2** [comment]
Wouldn't it take a lot of space? What does @agrogers think? Would you (@dbell154) like to implement this if this is accepted?

**Comment by agrogers** [comment]
Some docs have a relatively small amount of info. Others have a huge amount. I have rarely used that option so not sure what the info is. But good idea to make it more visible if it helps people know what the document is about. I suspect that could be helpful for maps. Text could be smaller than the rest of the text and limited to 256 chars perhaps.

**Comment by tuomas2** [comment]
Or maybe limit to 2 lines and use ellipsis. And by tapping that, you can get full info.

**Comment by agrogers** [comment]
Yep, long-click should always be a last resort I think.

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by dbell154** [comment]
I wasn't talking about including the information there,  primarily just the clickable information icon that opens that. 
The formatting of the provided information is too unpredictable to extract from.

**Comment by timbze** [comment]
1 idea: open the current AlertDialog (that shows the doc info) when clicking the document item row with a "Download" button displayed prominently at the top

**Comment by dbell154** [comment]
> Would you (@dbell154) like to implement this if this is accepted?

I still don't know enough about Kotlin and the way the code is organized to do it. I only get to look at it on the computer a few hours a week.
This is really all I wanted to do.
![DocInfo](https://user-images.githubusercontent.com/3357697/136462923-e618216e-ee3f-475b-8c6a-5795a9b584bb.png)
Plus I haven't finished setting up the github repository in Android Studio.

**Comment by dbell154** [comment]
> 1 idea: open the current AlertDialog (that shows the doc info) when clicking the document item row with a "Download" button displayed prominently at the top

That's possible except that I think the AlertDialog is only designed for three options, Yes, No, Cancel. Also, that requires two taps, when the information icon in the list would only require one, and it would be more visible.

**Comment by timbze** [comment]
@dbell154 the AlertDialog can have a custom view with any layout, so it could have a button at the top, and then the text scrolling underneath so that the "Download" button stays visible.

**Comment by timbze** [comment]
> Plus I haven't finished setting up the github repository in Android Studio.

Depending on how much you've done with git or how experienced you are, you might like SmartGit. I use it almost exclusively for version control.

**Comment by tuomas2** [comment]
> SmartGit

Me too, but its' good idea to configure IDE also :-)

**Comment by dbell154** [comment]
Okay, I gave it a shot implementing this. And it works, except I have a couple of issues:
1. The showAbout function in CommonUtils.kt is apparently written to only be called by DocumentSelectionBase.kt, so I copied the source and modified it to run in DocumentDownloadItemAdapter.kt. Obviously we would want to modify the CommonUtils.kt version to handle calls from both files but I'm uncomfortable about changing any global type code.
2. I modified the document_list_item.xml so the information icon would have more space and linked it to the always visible progressbar instead of the recommended string display which doesn't always show.
3. I'm not sure how I should proceed using Android Studio. I'm guessing I should first Update Project and then Commit my changes and then Push.

**Comment by tuomas2** [comment]
> * The showAbout function in CommonUtils.kt is apparently written to only be called by DocumentSelectionBase.kt, so I copied the source and modified it to run in DocumentDownloadItemAdapter.kt. Obviously we would want to modify the CommonUtils.kt version to handle calls from both files but I'm uncomfortable about changing any global type code.

Why showAbout can't be called from outside of DocumentSelectionBase? I don't see any dependencies.

> * I modified the document_list_item.xml so the information icon would have more space and linked it to the always visible progressbar instead of the recommended string display which doesn't always show.

Sounds good.

> * I'm not sure how I should proceed using Android Studio. I'm guessing I should first Update Project and then Commit my changes and then Push.

Are you familiar with Git & Github pull requests (if not, you can read docs @ github)? You should commit changes and push to your fork and create a pull request towards our develop branch.

**Comment by dbell154** [comment]
> > * The showAbout function in CommonUtils.kt is apparently written to only be called by DocumentSelectionBase.kt, so I copied the source and modified it to run in DocumentDownloadItemAdapter.kt. Obviously we would want to modify the CommonUtils.kt version to handle calls from both files but I'm uncomfortable about changing any global type code.
> 
> Why showAbout can't be called from outside of DocumentSelectionBase? I don't see any dependencies.

When I try calling it  with the same code as in DocumentSelectionBase.kt I get an error with this line
            GlobalScope.launch(Dispatchers.Main) {CommonUtils.showAbout(this@DocumentSelectionBase, document) }
None of the context statements I tried would work. Since I've only used Java I'm not familiar with that format of the context reference. What would I use when calling from DocumentDownloadItemAdapter.kt?

> > * I'm not sure how I should proceed using Android Studio. I'm guessing I should first Update Project and then Commit my changes and then Push.
> 
> Are you familiar with Git & Github pull requests (if not, you can read docs @ github)? You should commit changes and push to your fork and create a pull request towards our develop branch.

**Comment by tuomas2** [comment]
> When I try calling it with the same code as in DocumentSelectionBase.kt I get an error with this line
> GlobalScope.launch(Dispatchers.Main) {CommonUtils.showAbout(this@DocumentSelectionBase, document) }
> None of the context statements I tried would work. Since I've only used Java I'm not familiar with that format of the context reference. What would I use when calling from DocumentDownloadItemAdapter.kt?

```kt
CommonUtils.showAbout(context, document)
```

If you need add val in front of context in getView for example, you need to change signature like this:

```kt
class DocumentDownloadItemAdapter(
    val context: Context,
    private val downloadControl: DownloadControl,
    private val recommendedDocuments: Ref<RecommendedDocuments>
) : ArrayAdapter<Book>(context, R.layout.document_list_item, ArrayList<Book>()
```

**Comment by dbell154** [comment]
> ```kotlin
> CommonUtils.showAbout(context, document)
> ```
That works. Thanks.

**Comment by tuomas2** [comment]
Fixed by #1870
